### PR TITLE
Test Sharkey against supported Go versions

### PR DESCRIPTION
**Pull request by mbyczkowski**
After bumping Go versions, the integration test failed, because CN in
the cert is being ignored with there's any SAN provided (not just DNS
SAN). To address this, I have added a DNS SAN to both client and server
certs (and also regenerated the root cert, although that wasn't
necessary).

In order to make the cert generation easier the next time, I have added
a simple bash script--regenerate_certs.sh.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ services:
   - docker
 
 go:
-  - 1.6
-  - 1.7
+  - '1.11.x'
+  - '1.x'
```
Comment (csstaub) [inline .travis.yml]: Do we care about these? I would just do 1.11

Comment (mbyczkowski) [reply .travis.yml]: Last 2 releases are supported Go. If we want only one version, we probably want 1.12.x then.

Comment (csstaub) [reply .travis.yml]: Oh right, 1.12 is already out. Let's do 1.11 and 1.12 then

File: test/tls/generate_certs.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+function cleanup() {
+  echo "Cleanup..."
+  rm -rf $DIR
+}
+
+trap cleanup EXIT
+
+# Generate new root cert
+certstrap --depot-path $DIR init --common-name "CertAuth" --expires "10 years"
```
Comment (mcpherrinm) [inline test/tls/generate_certs.sh]: DIR should be in quotes, in case mktemp puts it in a path with spaces.

**Pull request merged**